File: Algorithms/028.ImplementStrStr.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if (needle == "") {
            return 0;
        }
        int lenstr = haystack.size();
        int lensubstr = needle.size();
        if (lenstr < lensubstr) {
            return -1;
        }
        int len = lenstr - lensubstr;
        for (int i=0; i <= len; i++) {
            int j;
            for (j=0; j < lensubstr; j++) {
                if (haystack[i+j] != needle[j]) {
                    break;
                }
            }
            if (j == lensubstr) {
                return i;
            }
        }
        return -1;
    }
};

File: Algorithms/028.ImplementStrStr.py
class Solution:
    # @param haystack, a string
    # @param needle, a string
    # @return an integer
    def strStr(self, haystack, needle):
        i = 0
        while True:
            j = 0
            while True:
                if j == len(needle):
                    return i
                if i + j == len(haystack):
                    return -1
                if needle[j] != haystack[i+j]:
                    break
                j += 1
            i += 1

